runtime/gc_vlhgc/WriteOnceCompactor.cpp

```diff
@@ -1204,6 +1206,51 @@ MM_WriteOnceCompactor::fixupMixedObject(MM_EnvironmentVLHGC* env, J9Object *obje
 	}
 }
 
+void
+MM_WriteOnceCompactor::doStackSlot(MM_EnvironmentVLHGC *env, J9Object *fromObject, J9Object** slot)
+{
+	/* ensure that this isn't a slot pointing into the gap (only matters for split heap VMs) */
+	if (!_extensions->heap->objectIsInGap(*slot)) {
```

Comment:
I guess you inherited this from generic RootScanner, but this is not a concern for any Balanced GC related doStackSlot. You can remove this condition safely.